### ISSUE TITLE
Improve unit test DX with test utilities and documentation

### ISSUE THREAD

**Issue by ahnpnl** [robisim74/qwik-speak#20]
## Issue

Currently there is no documentation how to unit test a Qwik component using `qwik-speak` and no test utilities either.

## Example unit tests

Reference https://dev.to/manuartero/testing-a-react-context-provider-5cfg

- Given a component `TestComponent.tsx` like below
```
import { component$ } from '@builder.io/qwik';
import { $translate } from 'qwik-speak';

export const TestComponent = component$(() => {
    return <h1>{$translate('banana_key')}</h1>;
});
```

- Given `speak-config.ts` like below
```
import { $ } from '@builder.io/qwik';
import { isServer } from '@builder.io/qwik/build';
import type { LoadTranslationFn, SpeakConfig, TranslationFn } from 'qwik-speak';

export const speakConfig: SpeakConfig = {
    defaultLocale: { lang: 'nl-NL', currency: 'EUR', timeZone: 'Europe/Amsterdam' },
    supportedLocales: [
        { lang: 'nl-NL', currency: 'EUR', timeZone: 'Europe/Amsterdam' },
        { lang: 'en-US', currency: 'EUR', timeZone: 'Europe/Amsterdam' },
    ],
    assets: ['common'],
};

export const loadTranslation$: LoadTranslationFn = $(async (lang: string, asset: string, origin?: string) => {
    if (import.meta.env.DEV) {
        let url = '';
        // Absolute urls on server
        if (isServer && origin) {
            url = origin;
        }
        url += `/assets/i18n/${lang}/${asset}.json`;
        const data = await fetch(url);

        return data.json();
    }

    return Promise.resolve();
});

export const translationFn: TranslationFn = {
    loadTranslation$,
};
```

- We have a unit test 
```
it('should translate', async () => {
        const Wrapper = component$(() => {
            /**
             * There are 2 choices here:
             * - Either developers would provide a stub loadTranslationFn
             * - Or in `speak-config.ts`, developers need to add extra logic to ensure translations are loaded in test environment
             */
            const loadTranslationStub$ = $(() => {
                // We can make this to return a JSON file as well
                return Promise.resolve({
                    banana_key: 'banana',
                });
            });

            return (
                <QwikSpeak
                    config={speakConfig}
                    translationFn={{
                        loadTranslation$: loadTranslationStub$,
                    }}
                >
                    <TestComponent />
                </QwikSpeak>
            );
        });
        const { screen, render } = await createDOM();

        await render(<Wrapper />);

        expect(screen.querySelector('h1')?.textContent).toBe('banana');
    });
```

### Explanation

- We don't mock context provider but we simply provide a wrapper around the component which uses `qwik-speak`. The wrapper component would be like `Wrapper` in the example.
- To ensure translations are loaded, developers need to provide proper implementation for `loadTranslation$` property of `translationFn`, see the example above.

## Suggestion

From the discussion at https://github.com/robisim74/qwik-speak/commit/f7d5920d787b2dd42d6b77f2fbae705fa5c2c374#r98961952, I think there should be:
- documentation added for unit test 
- test utilities

Maybe `qwik-speak` can expose a wrapper component for unit test which provides:
- Possibility to have children component so that any components use `qwik-speak` can be the children of it which allows them to access `SpeakContext`
- A recommended example to define `loadTranslation$` in such a way that it would work for dev, prod as well as test

**Comment by robisim74** [comment]
@ahnpnl Thanks a lot for this suggestion!

There is a real need for test utilities and related documentation. The idea of the wrapper component is simple and perfect as I like it.

I'll do some experimenting to figure out what's the simplest way on the DX side to provide loading of translations during testing, without interfering with dev and prod environment.

In the meantime I released a patch (v.0.5.1) which fixes the `document is not defined` error.

**Comment by robisim74** [comment]
I created a branch [Testing](https://github.com/robisim74/qwik-speak/tree/testing) where I added the unit test of a component: https://github.com/robisim74/qwik-speak/blob/testing/src/app/routes/%5B...lang%5D/index.spec.tsx

I also added the documentation: https://github.com/robisim74/qwik-speak/blob/testing/docs/testing.md

After some tests, I found that the current` QwikSpeak `component (as in v0.5.1) also works properly as wrapper in the test environment. So I just renamed it to the more suitable `QwikSpeakProvider`.

Any feedback or contribution is welcome.

**Comment by ahnpnl** [comment]
I like the new name, it makes it more familiar and closer to React Context pattern.

I checked the documentation as well, LGTM. Thanks a lot!

**Comment by robisim74** [comment]
Thank you for pointing me clearly in the right direction!

I will merge the branch soon in the next version.

**Comment by robisim74** [comment]
Implemented in v0.6.0
